pkg/doc/util/time.go

```diff
@@ -0,0 +1,140 @@
+package util
+
+import (
+	"time"
+)
+
+// TimeWithTrailingZeroMsec overrides marshalling of time.Time. It keeps a format of initial unmarshalling
+// in case when date has zero a fractional second (e.g. ".000").
+// For example, time.Time marshals 2018-03-15T00:00:00.000Z to 2018-03-15T00:00:00Z
+// while TimeWithTrailingZeroMsec marshals to the initial 2018-03-15T00:00:00.000Z value.
+type TimeWithTrailingZeroMsec struct {
+	time.Time
+
+	trailingZerosMsecCount int
+}
+
+// NewTime creates TimeWithTrailingZeroMsec without zero sub-second precision.
+// It functions as a normal time.Time object.
+func NewTime(t time.Time) *TimeWithTrailingZeroMsec {
+	return &TimeWithTrailingZeroMsec{Time: t}
+}
+
+// NewTimeWithTrailingZeroMsec creates TimeWithTrailingZeroMsec with certain zero sub-second precision.
+func NewTimeWithTrailingZeroMsec(t time.Time, trailingZerosMsecCount int) *TimeWithTrailingZeroMsec {
+	return &TimeWithTrailingZeroMsec{
+		Time:                   t,
+		trailingZerosMsecCount: trailingZerosMsecCount,
```

Comment:
just a thought:  preserving actual string value could have made logic more simpler

Reply:
I think the current solution is more generic - it keeps the format which can be obtained separately.